### ISSUE TITLE
Additional configuration for "incoming" attachment directory (with finn/signald docker container)

### ISSUE THREAD

**Issue by tobymurray** [mautrix/signal#14]
So far as I understand, the bridge/signal deployment is intended to share the attachment directory, so there's only ever one copy of the attachments. If that's incorrect, I imagine most of what follows will be as well:

I tried to use a docker-compose deployment making use of the finn/signald docker container ([Dockerfile](https://gitlab.com/thefinn93/signald/-/blob/master/Dockerfile) and [image](https://hub.docker.com/r/finn/signald)

```
services:
  mautrix-signal:
    container_name: mautrix-signal
    image: dock.mau.dev/tulir/mautrix-signal
    restart: unless-stopped
    volumes:
    - ./bridge:/data
    - ./signald:/signald
    depends_on:
    - signald
  signald:
    container_name: signald
    image: finn/signald
    restart: unless-stopped
    volumes: 
    - ./signald:/signald
```

One of the more significant differences between the two signald docker containers appears to be the directory where attachments are stored (`/signald/attachments` vs `/var/run/signald/attachments`). So far as I can tell, the mautrix-signal configuration for signald can only be used if the attachment directory has the same path in both the bridge container and the signald container. E.g. with the above docker-compose file and an `outgoing_attachment_dir: /signald/attachments`, that path is used when querying signald:
```
signald           | 14:42:15.163 [socketlistener] ERROR io.finn.signald.SocketHandler - Catching
signald           | org.asamk.signal.AttachmentInvalidException: /signald/attachments/mautrix-signal-ca29c1ae-fc0f-48a4-a0a7-64f3fe16587d: /signald/attachments/mautrix-signal-ca29c1ae-fc0f-48a4-a0a7-64f3fe16587d (No such file or directory)
signald           | 	at io.finn.signald.SocketHandler.send(SocketHandler.java:255) ~[signald.jar:unspecified]
signald           | 	at io.finn.signald.SocketHandler.handleRequest(SocketHandler.java:137) [signald.jar:unspecified]
signald           | 	at io.finn.signald.SocketHandler.run(SocketHandler.java:106) [signald.jar:unspecified]
signald           | 	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_272]
mautrix-signal    | [2020-11-17 14:42:15,164] [ERROR@mau.as] Exception in Matrix event handler
mautrix-signal    | Traceback (most recent call last):
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix/appservice/as_handler.py", line 178, in try_handle
mautrix-signal    |     await handler_func(event)
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/matrix.py", line 439, in int_handle_event
mautrix-signal    |     await self.handle_message(evt.room_id, evt.sender, evt.content, evt.event_id)
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/matrix.py", line 282, in handle_message
mautrix-signal    |     await portal.handle_matrix_message(sender, message, event_id)
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix_signal/portal.py", line 193, in handle_matrix_message
mautrix-signal    |     await self.signal.send(username=sender.username, recipient=self.chat_id, body=text,
mautrix-signal    |   File "/opt/mautrix-signal/mausignald/signald.py", line 117, in send
mautrix-signal    |     await self.request("send", "send_results", username=username, messageBody=body,
mautrix-signal    |   File "/opt/mautrix-signal/mausignald/rpc.py", line 157, in request
mautrix-signal    |     resp_type, resp_data = await self._raw_request(command, **data)
mautrix-signal    |   File "/opt/mautrix-signal/mausignald/rpc.py", line 154, in _raw_request
mautrix-signal    |     return await future
mautrix-signal    | mausignald.errors.UnexpectedError: /signald/attachments/mautrix-signal-ca29c1ae-fc0f-48a4-a0a7-64f3fe16587d: /signald/attachments/mautrix-signal-ca29c1ae-fc0f-48a4-a0a7-64f3fe16587d (No such file or directory)
```

If I change the configuration to match the signald container, signald is happy but the bridge is not:
```
mautrix-signal    | [2020-11-18 02:43:39,886] [ERROR@mau.as] Exception in Matrix event handler
mautrix-signal    | Traceback (most recent call last):
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix/appservice/as_handler.py", line 178, in try_handle
mautrix-signal    |     await handler_func(event)
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/matrix.py", line 439, in int_handle_event
mautrix-signal    |     await self.handle_message(evt.room_id, evt.sender, evt.content, evt.event_id)
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/matrix.py", line 282, in handle_message
mautrix-signal    |     await portal.handle_matrix_message(sender, message, event_id)
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix_signal/portal.py", line 188, in handle_matrix_message
mautrix-signal    |     attachment_path = await self._download_matrix_media(message)
mautrix-signal    |   File "/usr/lib/python3.8/site-packages/mautrix_signal/portal.py", line 162, in _download_matrix_media
mautrix-signal    |     with open(path, "wb") as file:
mautrix-signal    | FileNotFoundError: [Errno 2] No such file or directory: '/var/run/signald/attachments/mautrix-signal-cf04e03d-0858-41df-a475-91f3de0c51b7'
```

If I understand this correctly, it would be nice two have two configurations to support a docker-compose deployment: one to specify the path relative to the mautrix-signal container, and one to request from signald. This would enable the mautrix-signal and signald docker containers to put attachments wherever is convenient in the context of the docker container and still be able to communicate.

**Comment by tobymurray** [comment]
As a workaround, in my finn/signald container I've created a link so the paths can be treated the same between the bridge and signald, and I think that works.

**Comment by rufflepot** [comment]
Was this fixed by [this recent finn/signald commit](https://gitlab.com/thefinn93/signald/-/commit/cacff157cfa996af3389023cd8150b5409d562da)?

**Comment by tobymurray** [comment]
I think "worked around" is more accurate than "fixed", I commented #15 :
> With https://gitlab.com/thefinn93/signald/-/merge_requests/34, the finn/signald image has been aligned with dock.mau.dev/maunium/signald in regards to data directory. The issue this PR addresses should no longer be a concern, though theoretically it's still valid (works by "coincidence" presently, relying on the same paths in two distinct filesystems).
